pymongo/topology_description.py

```diff
@@ -324,6 +324,14 @@ def apply_selector(
             description = self.server_descriptions().get(address)
             return [description] if description else []
 
+        # Primary selection fast path.
+        if self.topology_type == TOPOLOGY_TYPE.ReplicaSetWithPrimary and type(selector) is Primary:
+            for sd in self._server_descriptions.values():
+                if sd.server_type == SERVER_TYPE.RSPrimary:
+                    return [sd]
```

Comment:
Oh one more thing, don't we have to call the `custom_selector` here if one was provided?

Reply:
And add a test for it?

Reply:
I'll have to defer for now, working on a HELP ticket

Reply:
Done